File: backend/src/main/java/com/gittowork/domain/coverletter/entity/CoverLetterAnalysis.java
package com.gittowork.domain.coverletter.entity;

import com.gittowork.domain.user.entity.User;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "cover_letter_analysis")
public class CoverLetterAnalysis {

    @Id
    @Size(max = 255)
    @Column(name = "cover_letter_analysis_id", nullable = false)
    private String coverLetterAnalysisId;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "file_id", nullable = false)
    private CoverLetter file;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Lob
    @Column(name = "analysis_result")
    private String analysisResult;

    @Column(name = "global_capability")
    private Integer globalCapability;

    @Column(name = "challenge_spirit")
    private Integer challengeSpirit;

    @Column(name = "sincerity")
    private Integer sincerity;

    @Column(name = "communication_skill")
    private Integer communicationSkill;

    @Column(name = "achievement_orientation")
    private Integer achievementOrientation;

    @Column(name = "responsibility")
    private Integer responsibility;

    @Column(name = "honesty")
    private Integer honesty;

    @Column(name = "creativity")
    private Integer creativity;

    @Column(name = "create_dttm")
    private LocalDateTime createDttm;

}

File: bigdata/app/models/cover_letter_analysis.py
from sqlalchemy import Column, String, Integer, TIMESTAMP, ForeignKey, Text
from sqlalchemy.orm import relationship
from app.core.database import Base

class CoverLetterAnalysis(Base):
    __tablename__ = "cover_letter_analysis"

    cover_letter_analysis_id = Column(String(255), primary_key=True)
    file_id = Column(Integer, ForeignKey("cover_letter.file_id"), nullable=False)
    user_id = Column(Integer, ForeignKey("user.user_id"), nullable=False)
    analysis_result = Column(Text, nullable=True)
    global_capability = Column(Integer, nullable=True)
    challenge_spirit = Column(Integer, nullable=True)
    sincerity = Column(Integer, nullable=True)
    communication_skill = Column(Integer, nullable=True)
    achievement_orientation = Column(Integer, nullable=True)
    responsibility = Column(Integer, nullable=True)
    honesty = Column(Integer, nullable=True)
    creativity = Column(Integer, nullable=True)
    create_dttm = Column(TIMESTAMP, nullable=True)

    # 관계
    cover_letter = relationship("CoverLetter", back_populates="analyses")
